### PR TITLE
Start newsletter#11

### PR DESCRIPTION
**Pull request by sunjon**
A rough draft attempt to cover some of the 0.5 features/timeline; LSP, Tree-sitter, vim-conf, etc.

There's no detail yet on the highlighted changes section or in general, and there's some overlap in what's already covered in newsletter#10/4.4.

### CODE REVIEW

File: _posts/2020-09-22-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+ - ext_popupmenu: completion menu
+ - ext_tabline: tab line
+ - ext_cmndline: command line
+ - ext_hlstate: highlight state
+ - ext_messages: messages
+ - ext_multigrid: windows, grids
+ - remote TUI (GSoC 2019)
+
+## Notable commits/merges
+libuv / libvrwem ?
```
Comment (bfredl) [inline _posts/2020-09-22-newsletter.md:200]: libvterm?

**Comment by tjdevries** [comment]
For LSP & Lua, you could link directly to my talks about them from vimconf, as I think they give good info :laughing: 

I can put the links here if you want them.

File: _posts/2020-09-22-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+nvim__screenshot?
+
+## PR's to watch
+
+buf_set_text ? what will it enable?
+### #12249 https://github.com/neovim/neovim/pull/12249 nvim_buf_set_text
+
+### #12870 https://github.com/neovim/neovim/pull/12870 Color themes per window/line
+### #12823 https://github.com/neovim/neovim/pull/12823 Doc ; Lua concepts
+### #12593 https://github.com/neovim/neovim/pull/12593 fswatrch autoread
```
Comment (vigoux) [inline _posts/2020-09-22-newsletter.md:214]:
```suggestion
### #12593 https://github.com/neovim/neovim/pull/12593 fswatch autoread
```

File: _posts/2020-09-22-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+### missing functionality for Lua developers
+
+### #12378 https://github.com/neovim/neovim/pull/12378 [WIP] lua: Add autocmd with Lua callback natively #12378
+
+## Merrged (needs more thorough review of PR's from 0.4 -> 0.5)
```
Comment (vigoux) [inline _posts/2020-09-22-newsletter.md:220]:
```suggestion
## Merged (needs more thorough review of PR's from 0.4 -> 0.5)
```

File: _posts/2020-09-22-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+"A really awesome parser"
+"Tree sitter offers a way to navigate ASTs"
+"tree-sitter queries are the new snippets"?
+Error recovery
+"power of vim editing == textobjects" and "tree-sitter == smarter textobjects"
+
+A presentation from the author on what Tree-sitter is and how it came to be:
+[Tree-sitter - A new parsing system for programming tools - Max Brunsfield](https://www.youtube.com/watch?v=Jes3bD6P0To&feature=youtu.be&t=232)
+
+@norcalli - "every file you edit (which has a treesitter grammar definition available) will have the semantic information of what you're working on available to use for anything you want. It's like a core intelligence was added to neovim which was fundamentally missing before."
```
Comment (vigoux) [inline _posts/2020-09-22-newsletter.md:44]: While I agree with what this quote says, i'd just correct that to : 
```suggestion
@norcalli - "every file you edit (which has a treesitter grammar definition available) will have the syntactic information of what you're working on available to use for anything you want. It's like a core intelligence was added to neovim which was fundamentally missing before."
```

File: _posts/2020-09-22-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+### Core features being implemented in Lua
+
+- [neovim-lspconfig](https://github.com/neovim/nvim-lspconfig) - Collection of common configurations for the Nvim LSP client.
+- [completion-nvim](https://github.com/nvim-lua/completion-nvim)  - An auto completion framework that aims to provide a better completion experience with neovim's built-in LSP.
+- [diagnostic-nvim](https://github.com/nvim-lua/diagnostic-nvim)  - A wrapper for the default LSP diagnostics aimed to make for a more user friendly setup.
+- [lsp-status.nvim](https://github.com/nvim-lua/lsp-status.nvim)  - A Neovim plugin/library for generating statusline components from the built-in LSP client.
+
+### Vimscript vs Lua
+
```
Comment (clason) [inline _posts/2020-09-22-newsletter.md:97]: mention https://github.com/nanotee/nvim-lua-guide here?

**Comment by sunjon** [comment]
> For LSP & Lua, you could link directly to my talks about them from vimconf, as I think they give good info laughing
> 
> I can put the links here if you want them.

Yes please, direct links to any of the content would be good (and any accompaying slides)

**Comment by tjdevries** [comment]
Builtin LSP: https://youtu.be/C9X5VF9ASac // Slides: https://github.com/tjdevries/config_manager/blob/master/presentations/neovim_lsp/presentation.md

Why Lua: https://youtu.be/IP3J56sKtn0?t=21 // Slides: https://github.com/tjdevries/config_manager/blob/master/presentations/lua_design_principles/presentation.md

**Comment by teto** [comment]
@sunjon Great initiative. Note that we are open to other type of blog posts than just the release ones, so anyone should feel free to propose a blog post focusing on either LSP or tree-sitter etc.

File: _posts/2020-09-22-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
+---
+layout: newsletter
+title: "Newsletter #11 - **AWESOME TITLE HERE**
```
Comment (teto) [inline _posts/2020-09-22-newsletter.md:3]: "halfway there" (with respect to the 1.0) :)

**Comment by teto** [comment]
Ideally I would envision 2 companion blog posts: about LSP and treesitters. These are brand new introductions and too complex to mix with a newsletter (introduction + technical setup + current state/ecosystem and what's next). These could serve as the official `soft` reference (in contrast to `:help ` or `nvim-lspconfig`). Then there is always the matter of writing those. I believe @vigoux might have something for treesitter ? as for LSP, if there is no volunteer I can come up with something.

**Comment by vigoux** [comment]
I have been doing some things, but nothing significant yet, I'll try to have it ready soon.

**Comment by teto** [comment]
I've put @sunjon 's template on https://github.com/neovim/neovim/wiki/newsletter-%2311-:-the-0.5-release so that we can collaboratively edit it.

**Comment by rockerBOO** [comment]
What is currently needed to push this? Looks like some sections need to be finished up.

I think the [0.5 release commit](https://github.com/neovim/neovim/commit/a5ac2f45ff84a688a09479f357a9909d5b914294) has a good overview of what has changed as well.

**Comment by bfredl** [comment]
probably it would make sense to move the too unfinished parts to a new wip newsletter `#11` and finish off the parts relevant to the release (and yes some stuff could be copied over from the release commit).

**Comment by clason** [comment]
and it's published: https://neovim.io/news/

thanks @sunjon for the first draft, that was very helpful!

**Pull request closed without merging**